src/qz/utils/PrintingUtilities.java

```diff
@@ -209,7 +210,15 @@ public static void processPrintRequest(Session session, String UID, JSONObject p
             }
 
             processor.parseData(params.getJSONArray("data"), options);
-            processor.print(output, options);
+
+            try {
```

Comment:
I don't like the nested try, but I couldn't find a way to query if a PrintService was still valid

Reply:
Agreed.  I was able to avoid the nested try, but I haven't tested it.  The code is here: https://github.com/qzind/tray/compare/master...tresf:tray:ps-avail